### PR TITLE
repair: ensure max single repair job per host

### LINKED ISSUE #3425: `--parallel` flag to ensure single repair job per host
Current definition of `--parallel` flag is as follow https://manager.docs.scylladb.com/stable/sctool/repair.html#parallel

> The maximum number of Scylla repair jobs that can run at the same time (on different token ranges and replicas). Each node can take part in at most one repair at any given moment. By default the maximum possible parallelism is used. The effective parallelism depends on a keyspace replication factor (RF) and the number of nodes. The formula to calculate is is as follows: number of nodes / RF, ex. for 6 node cluster with RF=3 the maximum parallelism is 2.

We have to ensure that only a single repair job is executed on a host. Definition of the `--parallel` flag stays as-is.

~~We want to change it to~~
~~> The maximum number of Scylla repair jobs that can run at the same time on a single host.~~

~~Given a scenario where:~~
~~- there are 4 nodes A, B, C, D~~
~~- replication factor is set to 2~~
~~- woth ranges r1, r2, r3, r4, r5, r6~~
~~- ranges are distributed as follow~~
~~r1 are on nodes A, B~~
~~r2 -> A, C~~
~~r3 -> A. D~~
~~r4 -> B, C~~
~~r5 -> B, D~~
~~r6 -> C, D~~
~~.... with parallel set to 1 only following combinations are possible...~~
~~r1(A, B) together with r6(C, D)~~
~~r2(A, C) together with r5(B, D) etc~~

~~We must assure that max `--parallel` number of jobs are executed in parallel on a single host~~

**Default for `parallel`: 1**

### PR DESCRIPTION
**Pull request by karol-kokoszka**
Fixes #3425 

---

"Only the Best is Good Enough." - LEGO company motto

Please make sure that:
- Code is split to commits that address a single change
- Commit messages are informative
- Commit titles have module prefix
- Commit titles have issue nr. suffix

### CODE REVIEW

File: pkg/service/repair/controller.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ func (c *rowLevelRepairController) shouldBlock(hosts []string, intensity int) bo
 
 	// DENY if any host has max nr of jobs already running
 	for _, h := range hosts {
-		if c.jobs[h] >= c.limits[h].Default {
+		if c.jobs[h] >= 1 {
```
Comment (mykaul) [inline pkg/service/repair/controller.go:77]: nit: the comment in line 75 is now inaccurate.

Comment (karol-kokoszka) [reply pkg/service/repair/controller.go:66]: thanks, updated the comment + removed another checks that are not relevant anymore

File: pkg/service/repair/controller_test.go [hunk 2]
```diff
@@ -199,44 +155,3 @@ func TestRowLevelRepairController(t *testing.T) {
 		}
 	})
 }
-
-func TestRowLevelRepairControllerIssue2446(t *testing.T) {
```
Comment (vladzcloudius) [inline pkg/service/repair/controller_test.go:203]: Why this test was removed?

File: pkg/service/repair/controller_test.go [hunk 3]
```diff
@@ -47,125 +47,81 @@ func (s controllerTestSuite) newRowLevelRepairController() *rowLevelRepairContro
...
-		if ok, _ := ctl.TryBlock([]string{"a", "b", "c"}); !ok {
+		const intensity = 1000
+
+		s.intensityHandler.SetIntensity(ctx, intensity)
+
+		ok, a := ctl.TryBlock([]string{"a", "b", "c"})
+		if !ok {
 			t.Fatal("TryBlock() failed to block")
 		}
+		if a.Ranges != intensity {
```
Comment (vladzcloudius) [inline pkg/service/repair/controller_test.go:62]: Should be ">" - exactly as the error message below states.
Repair task is allowed to pick less than "intensity" ranges, e.g. in the last hunk.

**Review by vladzcloudius — CHANGES_REQUESTED** [review]

**Comment by vladzcloudius** [comment]
There is already a PR that fixes this and other issues which was submitted some time ago: https://github.com/scylladb/scylla-manager/pull/3506

This PR pretty much copies what this patch (https://github.com/scylladb/scylla-manager/pull/3506/commits/8020beb4db314e598ae1f42d83b2e405a3e892d9) from the PR above does in terms of SM fix.

Don't you think that it makes more sense to pick the code part from the PR above and leave only the test part from this PR?

**Comment by vladzcloudius** [comment]
> This PR pretty much copies what this patch (https://github.com/scylladb/scylla-manager/commit/8020beb4db314e598ae1f42d83b2e405a3e892d9) from the PR above does in terms of SM fix.

I need to take this back - it didn't. It should have though ;)

**Comment by karol-kokoszka** [comment]
Closing as https://github.com/scylladb/scylla-manager/pull/3521 is ready

**Pull request closed without merging**